pkg/awsiamauth/config/aws-iam-authenticator.yaml

```diff
@@ -0,0 +1,202 @@
+        operator: Exists
+
+      # `aws-iam-authenticator server` has four volumes
+      # - config (mounted from the ConfigMap at /etc/aws-iam-authenticator/config.yaml)
+      # - cert, key (persisted TLS certificate and key, mounted from the host)
+      # - kubeconfig (kubeconfig to plug into your apiserver configuration, mounted from the host)
+      containers:
+      - name: aws-iam-authenticator
+        image: {{.image}}
+        env:
+        - name: AWS_REGION
+          value: {{.awsRegion}}
+        args:
+        - server
+        - --backend-mode={{.backendMode}}
+        - --partition={{.partition}}
+        - --config=/etc/aws-iam-authenticator/config.yaml
+        - --state-dir=/var/aws-iam-authenticator
+        - --generate-kubeconfig=/etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml
+        - --kubeconfig-pregenerated=true
+
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - ALL
+
+        resources:
+          requests:
+            memory: 20Mi
+            cpu: 10m
+          limits:
+            memory: 20Mi
+            cpu: 100m
+
+        volumeMounts:
+        - name: config
+          mountPath: /etc/aws-iam-authenticator/
+        - name: cert
+          mountPath: /var/aws-iam-authenticator/cert.pem
+        - name: key
+          mountPath: /var/aws-iam-authenticator/key.pem
+        - name: kubeconfig
+          mountPath: /etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml
+
+      # init container to set permissions for the mount paths
+      initContainers:
+      - name: chown
+        image: busybox
+        command: ['sh', '-c', 'chown 10000:10000 /var/aws-iam-authenticator/cert.pem; chown 10000:10000 /var/aws-iam-authenticator/key.pem; chown 10000:10000 /etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml']
+        volumeMounts:
+        - name: cert
+          mountPath: /var/aws-iam-authenticator/cert.pem
+        - name: key
+          mountPath: /var/aws-iam-authenticator/key.pem
+        - name: kubeconfig
+          mountPath: /etc/kubernetes/aws-iam-authenticator/kubeconfig.yaml
+
+      volumes:
+      - name: config
+        configMap:
+          name: aws-iam-authenticator
+      - name: kubeconfig
+        hostPath:
+          path: /var/lib/kubeadm/aws-iam-authenticator/kubeconfig.yaml
+      - name: cert
+        hostPath:
+          path: /var/lib/kubeadm/aws-iam-authenticator/pki/cert.pem
+      - name: key
+        hostPath:
+          path: /var/lib/kubeadm/aws-iam-authenticator/pki/key.pem
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  namespace: kube-system
+  name: aws-iam-authenticator
+  labels:
+    k8s-app: aws-iam-authenticator
+data:
+  config.yaml: |
+    clusterID: {{.clusterID}}
+
+---
+# EKS-Style ConfigMap: roles and users can be mapped in the same way as supported on EKS.
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: aws-auth
+  namespace: kube-system
+data:
+  mapRoles: |
```

Comment:
should the `{{- if (ne .mapRoles "")}}` check be moved before `mapRoles: |` section starts? same for mapUsers

Reply:
Sure 👍  
I just kept it like that in case a user was to describe the cm they would see that they set these keys as empty.